src/model_repository_manager.h

```diff
@@ -72,12 +73,73 @@ class ModelRepositoryManager {
     const std::string reason_;
   };
 
+  // Information about the model.
+  struct ModelInfo {
+    ModelInfo(
+        const int64_t mtime_nsec, const int64_t prev_mtime_ns,
+        const std::string& model_path)
+        : mtime_nsec_(mtime_nsec), prev_mtime_ns_(prev_mtime_ns),
+          explicitly_load_(true), model_path_(model_path),
+          is_config_provided_(false)
+    {
+    }
+    ModelInfo()
+        : mtime_nsec_(0), prev_mtime_ns_(0), explicitly_load_(true),
+          is_config_provided_(false)
+    {
+    }
+    int64_t mtime_nsec_;
+    int64_t prev_mtime_ns_;
+    bool explicitly_load_;
+    inference::ModelConfig model_config_;
+    std::string model_path_;
+    // Temporary location to hold agent model list before creating the model
+    // the ownership must transfer to ModelLifeCycle to ensure
+    // the agent model life cycle is handled properly.
+    std::shared_ptr<TritonRepoAgentModelList> agent_model_list_;
+    bool is_config_provided_;
+  };
+
+  // Map from model name to information about the model.
+  struct ModelInfoMap {
+    ModelInfoMap() = default;
+    ModelInfoMap(const ModelInfoMap& rhs);
+    ModelInfoMap& operator=(const ModelInfoMap& rhs);
+
+    void Swap(ModelInfoMap& rhs);
+    Status GetModelInfo(
+        const ModelIdentifier& model_id, ModelInfo** model_info) const;
+
+    // Write updated model info back to this object after model load/unload.
+    void Writeback(
+        const ModelInfoMap& updated_model_info,
+        const std::set<ModelIdentifier>& affected_models);
+
+    std::unordered_map<ModelIdentifier, std::unique_ptr<ModelInfo>> map_;
+  };
+
+  // std::condition_variable wrapper with mutex included
+  class ConditionVariable {
```

Comment:
Give it a different name, it is easy to be confused with `std::condition_variable`, and the use of this class doesn't seem to be reflected by the name, i.e. it is not how I would imagine to use when I see a "condition variable".

Reply:
Update commit: Rename ConditionVariable to EventNotifier

Reply:
The ConditionVariable / EventNotifier / event listener intermediate class is no longer necessary with some logic simplification.
Update commit: Refactor conflict notify mechanism